homeassistant/components/vizio/media_player.py

```diff
@@ -106,127 +141,160 @@ def __init__(
         self._supported_commands = SUPPORTED_COMMANDS[device_type]
         self._device = device
         self._max_volume = float(self._device.get_max_volume())
-        self._unique_id = None
+        self._unique_id = unique_id
         self._icon = ICON[device_type]
+        self._available = True
 
     @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
     async def async_update(self) -> None:
         """Retrieve latest state of the device."""
+        is_on = await self._device.get_power_state(False)
+
+        if is_on is None:
+            self._available = False
+            return
+
+        self._available = True
 
         if not self._unique_id:
-            self._unique_id = await self._device.get_esn()
+            self._unique_id = await self._device.get_esn(False)
 
-        is_on = await self._device.get_power_state()
+        if not is_on:
+            self._state = STATE_OFF
+            self._volume_level = None
+            self._current_input = None
+            self._available_inputs = None
+            return
 
-        if is_on:
-            self._state = STATE_ON
+        self._state = STATE_ON
 
-            volume = await self._device.get_current_volume()
-            if volume is not None:
-                self._volume_level = float(volume) / self._max_volume
+        volume = await self._device.get_current_volume(False)
+        if volume is not None:
+            self._volume_level = float(volume) / self._max_volume
 
-            input_ = await self._device.get_current_input()
-            if input_ is not None:
-                self._current_input = input_.meta_name
+        input_ = await self._device.get_current_input(False)
+        if input_ is not None:
+            self._current_input = input_.meta_name
 
-            inputs = await self._device.get_inputs()
-            if inputs is not None:
-                self._available_inputs = [input_.name for input_ in inputs]
+        inputs = await self._device.get_inputs(False)
+        if inputs is not None:
+            self._available_inputs = [input_.name for input_ in inputs]
 
-        else:
-            if is_on is None:
-                self._state = None
-            else:
-                self._state = STATE_OFF
+    @staticmethod
+    async def _async_config_entry_updated(
```

Comment:
Since we use the dispatch helper, we could move this to the component and register the listener in `async_setup_entry` there.

It doesn't really matter here though since we have only one entity per entry. If we would have more than one entity, it would be more efficient to move, since we would only register one listener instead of many.

So either way is ok I think.

Reply:
I will address this if I do add another entity for an entry, but it's doubtful. I added a comment so I don't forget